nifi-nar-bundles/nifi-poi-bundle/nifi-poi-processors/src/main/java/org/apache/nifi/processors/poi/ConvertExcelToCSVProcessor.java

```diff
@@ -199,6 +213,10 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
         final int firstRow = context.getProperty(ROWS_TO_SKIP).evaluateAttributeExpressions(flowFile).asInteger() - 1;
         final List<Integer> columnsToSkip = getColumnsToSkip(context, flowFile);
 
+        // Set min inflate ratio before loading documents
+        final float minInflateRatio = context.getProperty(MIN_INFLATE_RATIO).evaluateAttributeExpressions(flowFile).asFloat();
+        ZipSecureFile.setMinInflateRatio(minInflateRatio);
```

Comment:
That's a good point, I don't know of any way to not make it static across classes, except maybe to have them have their own classloaders? Not sure that would work in this case. But I believe the ExcelReader will suffer from this too, so hopefully we can come up with something.

Reply:
One thing to consider though is if all the files are trusted, then if all the processors had the same setting for this property then it would be acceptable for the users' flows. We could perhaps check if the property value is more than the current setting and always keep the minimum. Definitely not a great user experience, just trying to give a way to prevent the Zip Bomb prediction from failing legitimate trusted files.

Reply:
Thanks for the reply. Yes, if all components use the same setting, that would work.

Checking the current value is a good idea. This could be checked during custom validation, and the property could be marked invalid if it is current minimum, using the default value of `0.01` to determine whether the value is set elsewhere.

Using instance class-loading is another interesting option. It is a bit of a heavy-weight solution in light of the optional property, but it may be the safest solution. I agree this would also impact the ExcelReader. Might be worth some additional evaluation of the supporting library classes.

Reply:
The hard part with checking the minimum is that if the first time the minimum was set lower, how can it be reset to the default for example? We wouldn't know what all the other instances are set to at that moment.